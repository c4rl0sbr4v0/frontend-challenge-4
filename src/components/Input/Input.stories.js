import React from 'react';
import Input from './index';

export default {
  title: 'Components/Input',
  component: Input,
};

export const Default = () => {
  const [input, setInput] = React.useState('');
  return (
    <Input
      onChange={(e) => setInput(e.target.value)}
      onKeyUp={(e) => {
        if (e.key === 'Enter') {
          e.preventDefault();
          alert(input);
        }
      }}
      value={input}
    />
  );
};
