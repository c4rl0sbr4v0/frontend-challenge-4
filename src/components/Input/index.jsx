import React from 'react';
import PropTypes from 'prop-types';

/**
 *  Input: Render an Input
 */
const Input = ({ value, onChange, onKeyUp, placeholder }) => {
  return (
    <input
      className="landbot-input"
      type="text"
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      onKeyUp={onKeyUp}
    />
  );
};

export default Input;

Input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onKeyUp: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};
Input.defaultProps = {
  value: '',
  onChange: undefined,
  onKeyUp: undefined,
  placeholder: 'Insert text here',
};
