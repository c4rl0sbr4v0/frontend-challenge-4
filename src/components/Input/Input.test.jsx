import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Input from '.';

test('Input Component', async () => {
  window.alert = jest.fn();
  const { container, getByPlaceholderText } = render(
    <Input onChange={() => alert('change')} value="patata" onKeyUp={() => alert('keyup')} placeholder="Enter" />,
  );
  const input = getByPlaceholderText('Enter');
  fireEvent.change(input, { target: { value: 'ey' } });
  fireEvent.keyUp(input, { key: 'Enter', code: 'Enter' });
  expect(window.alert).toHaveBeenCalledTimes(2);
  expect(container.firstChild.classList.contains('landbot-input')).toBe(true);
  expect(container).toMatchSnapshot();
});
