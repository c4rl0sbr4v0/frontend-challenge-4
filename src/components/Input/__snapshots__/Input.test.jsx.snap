// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component 1`] = `
<div>
  <input
    class="landbot-input"
    placeholder="Enter"
    type="text"
    value="patata"
  />
</div>
`;
